src/module.c

```diff
@@ -1685,7 +1686,15 @@ int RM_ReplyWithNull(RedisModuleCtx *ctx) {
  * execute some command, as we want to reply to the client exactly the
  * same reply we obtained by the command.
  *
- * The function always returns REDISMODULE_OK. */
+ * Return:
+ * * REDISMODULE_OK On success returns .
+ * * REDISMODULE_ERR if the given reply is in resp3 format but the client expects resp2
+ *   In case of error, its the module writer responsibility to translate the reply
+ *   to resp2 (or maybe handle it differently by returning an error). Notice that for
+ *   module writer convinient, it is possible to pass `0` as a parameter to the fmt
+ *   argument of `RM_Call` so that the RedisModuleCallReply will return in the same
+ *   format (resp2 or resp3) as the current client expects it.
```

Comment:
somewhere, salvatore had a long post about RESP3 being upper case and not a version thing (not RESP v3)
```suggestion
 * * REDISMODULE_ERR if the given reply is in RESP3 format but the client expects RESP2
 *   In case of error, it's the module writer responsibility to translate the reply
 *   to RESP2 (or maybe handle it differently by returning an error). Notice that for
 *   module writer convinient, it is possible to pass `0` as a parameter to the fmt
 *   argument of `RM_Call` so that the RedisModuleCallReply will return in the same
 *   format (RESP2 or RESP3) as the current client expects it.
```